bridges/snowbridge/pallets/system/src/lib.rs

```diff
@@ -635,6 +641,34 @@ pub mod pallet {
 				pays_fee: Pays::No,
 			})
 		}
+
+		/// Registers a Polkadot-native token as a wrapped ERC20 token on Ethereum.
+		/// Privileged. Can only be called by root.
+		///
+		/// Fee required: No
+		///
+		/// - `origin`: Must be root
+		/// - `location`: Location of the asset (relative to this chain)
+		/// - `metadata`: Metadata to include in the instantiated ERC20 contract on Ethereum
+		#[pallet::call_index(11)]
+		#[pallet::weight(T::WeightInfo::register_token())]
+		pub fn register_token_v2(
```

Comment:
Actually, ignore my comment, I realize we need separate register extrinsics for V1 and V2 as they have different fee-taking behavior.

I see now there is also the problem of how to provide WETH as the fees for these governance calls. On BH we can only provide DOT

Reply:
Maybe we need to make the rewards pallet support both WETH and DOT.

Reply:
> how to provide WETH as the fees for these governance calls.

Actually there is no fee charged for governance calls.

Reply:
The problem is that then no off-chain relayer will be incentivized to deliver the message. Letting users optionally pay using DOT is a solution we should consider.

Another issue is that when we make `register_token_v2` permissionless, there is still no way for many parachains to call the extrinsic since they won't have HRMP channels registered.

So I think we have to force users to route their governance calls through AH and provide DOT. In which case, BH will see message like:

```
AliasOrigin /Parachain/Hydra
ReceiveTeleportedAsset (DOT, 10)
DepositAsset (DOT, /Parachain/Hydra)
Transact(EthereumSystem.register_token_v2)
```

Reply:
> So I think we have to force users to route their governance calls through AH and provide DOT.

Agree. All calls should be through AH. 

But I'd assume in this case the fee could also be swapped to `WETH` before hand in AH so that in BH we only support WETH as fee.
https://github.com/yrong/polkadot-sdk/blob/cff1cba3dfc21d2be621eae984b83e4e0963c81b/bridges/snowbridge/primitives/router/src/outbound/v2/convert.rs#L289-L294

Reply:
Btw: this is the xcm format on BH
https://github.com/yrong/polkadot-sdk/blob/cff1cba3dfc21d2be621eae984b83e4e0963c81b/bridges/snowbridge/primitives/router/src/outbound/v2/convert.rs#L88-L96

Reply:
I would also prefer a solution where we only have 1 asset type in the rewards pallet.

Reply:
Yeah, that would be best.

If we move governance extrinsics such as register-token, create-agent to AH, then that would neatly solve the problem